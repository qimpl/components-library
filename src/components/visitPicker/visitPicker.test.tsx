import React from 'react';

import { render } from '@testing-library/react';

import VisitPicker from 'components/visitPicker';

const userAvailabilities = [
  { weekday: 0, startTime: '09:00', endTime: '11:00' },
  { weekday: 0, startTime: '14:00', endTime: '18:00' },
  { weekday: 1, startTime: '09:00', endTime: '11:00' },
  { weekday: 4, startTime: '02:00', endTime: '05:00' },
];

const userBookedVisits = [
  { hour: '10:00', date: '2021-04-11' },
  { hour: '14:00', date: '2021-04-11' },
];

test('Visit Picker test', () => {
  expect(
    render(
      <VisitPicker
        userAvailabilities={userAvailabilities}
        userBookedVisits={userBookedVisits}
        noAvailabilityMsg='No slot available for this date'
        handleClickVisitSlot={() => 'button click'}
      />
    )
  ).toMatchSnapshot();
});
