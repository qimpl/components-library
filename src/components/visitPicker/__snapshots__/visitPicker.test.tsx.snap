// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Visit Picker test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-iBPRYJ FpDPb"
      >
        <section>
          <div>
            <header>
              <h3
                class="sc-jSgupP dZQGQG"
                font-size="16px"
              >
                Sat Apr 10 2021
              </h3>
              <button
                class="sc-bdfBwQ fPgVyf"
                color="rgba(67,80,88,0.8)"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  focusable="false"
                  height="24"
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M14 4L12.9 5.1 18.9 11.2 2 11.2 2 12.8 18.9 12.8 12.9 18.9 14 20 22 12z"
                  />
                </svg>
              </button>
            </header>
            <section>
              <p
                class="sc-gsTCUz eNYSDA"
                style="color: rgb(1, 60, 99); font-size: 15px; line-height: 21px;"
              >
                No slot available for this date
              </p>
            </section>
          </div>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-iBPRYJ FpDPb"
    >
      <section>
        <div>
          <header>
            <h3
              class="sc-jSgupP dZQGQG"
              font-size="16px"
            >
              Sat Apr 10 2021
            </h3>
            <button
              class="sc-bdfBwQ fPgVyf"
              color="rgba(67,80,88,0.8)"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="24"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M14 4L12.9 5.1 18.9 11.2 2 11.2 2 12.8 18.9 12.8 12.9 18.9 14 20 22 12z"
                />
              </svg>
            </button>
          </header>
          <section>
            <p
              class="sc-gsTCUz eNYSDA"
              style="color: rgb(1, 60, 99); font-size: 15px; line-height: 21px;"
            >
              No slot available for this date
            </p>
          </section>
        </div>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
